web/docs/understand-vast/data-model/extractors.md

```diff
@@ -0,0 +1,76 @@
+# Extractors
+
+*Extractors* describe fields within a schema. The extractor resolution converts
+an extractor into an ordered list of *offsets*, which each uniquely describe a
+single (possibly nested) field in a given schema. The extractor resolution takes
+place once per schema. This enables a consistent way to specify which fields ot
+operate on throughout VAST across multiple schemas, e.g., in the [query
+language](/docs/understand-vast/query-language/) or the [index
+configuration](/docs/setup-vast/tune#tune-sketch-parameters).
+
+Let's look at how extractors work based on a subset of the `zeek.conn` schema:
+
+```yaml
+# builtins.yaml
```

Comment:
You could just use the module name here instead. I'm not sure if we agreed on a "global" namespace yet?

Reply:
It needs to be global if we want backwards compatibility, so in the interest of developer time we should have a global namespace rather than a `vast` namespace for the builtin types.

Reply:
I can see the argument, but I don't think we've agreed to prioritize backwards compatibility yet.

Reply:
I have yet to see a convincing argument _for_ breaking. The global namespace is consistent with our current naming. There's three ways to proceed here:

1. We use the global namespace as I suggested, and do not break anything.
2. We don't use a global namespace for built-in types and add a workaround for backwards compatibility. I see three options here currently:
   - We lazily add the `vast.` prefix when materializing table slices from the store, and when merging partition synopses in the catalog on startup.
   - We modify type equality comparison to no longer be just a call to `memcmp`, and instead do a deep equality comparison that accounts for this. Type hashing gets an additional branch to check for builtin types to keep the hashes consistent. Additionally, we would need to change the extractor resolution from this PR to account for the change.
   - We make a hard break and write a custom upgrade tool for the database.
3. We don't use a global namespace and make a hard break.

Of all these options only (1) sounds sensible to me given that we've managed to come by with a global namespace so far without any issues.

I think this discussion is very much relevant to this PR, and for anything other than option (1) I would want to see a convincing argument. Not just because of backwards compatibility, but also because of the development effort required for (2), given that we cannot afford to do (3) realistically.